foam_wrapper/foam_wrapper.py

```diff
@@ -0,0 +1,158 @@
+""" foam_wrapper module
+
+Wrapper module for OpenFOAM using pythonFlu -python interface
+NOTE: at a moment pythonFlu supports only up to 2.0.1 -version of OpenFOAM
+
+"""
+
+from Foam import ref, man, src
+from simphony.cuds.mesh import Mesh, Point, Face, Edge, Cell
+import inspect
+from Foam.OpenFOAM import fileName, word, IOobject
+from collections import defaultdict
+from Foam.finiteVolume import volScalarField
+from Foam.src.OpenFOAM import *
+
+class FoamWrapper(object):
+
+    def __init__(self, model):
+        self.model = model
+
+    def has_common_vertex(self, facePoints0, facePoints1):
+        for point in facePoints0:
+                if point in facePoints1:
+                    return True
+        return False
+
+    def get_mesh(self, foamCaseName):
+        """ Reads given OpenFoam -mesh from "foamCaseName" -directory and returns SimPhony mesh
+        """
+        argv=["test", "-case", foamCaseName]
+        args = ref.setRootCase(len(argv), argv)
+        runTime = man.createTime(args)
+        foamMesh = man.createMesh(runTime)
+
+        foamPoints = foamMesh.points()
+        foamFaces = foamMesh.faces()
+        foamCells = foamMesh.cells()
+
+        simphonyMesh = Mesh()
```

Comment:
Good 'point' there. I think there is not a common agreement how the wrapper hold information of the mesh and is it possible that it has multiple meshes. Looking at the proposed interface in wiki https://publicwiki-01.fraunhofer.de/SimPhoNy-Project/index.php/Wrapper_description  the name of the mesh in method get_mesh is given which gives the notion that there is multiple meshes in one wrapper instance (or wrapper class). 
Actually in the foam_wrapper packages current version the name  is the case name, not mesh name.  OpenFoam can't have meshes which have different names. Though every time step can have different mesh. 
We have  pushed to github current version of the foam_tokenwrapper, which is native c++ interface wrapper to OpenFoam variables (mesh,scalar and vector variables). Unfortunately there is not yet the get_mesh method, but the idea there was to give the OpenFoam arguments, i.e., the modeling engine state separately (initArgs method at a moment).
One thing is also should the wrapper hold the information of the uid mapping to original node labels. If it should, how the mapping is changed when new points are added in Simphony side. For example if we think that example piece of code you gave, then after add_point  method call there should be a call to change the mapping too. Of course there is multiple ways to overcome this (wrapper has add_point method, mesh import back to modeling engine recognise points which are not mapped,...).